### ISSUE TITLE
Test stays in idle state

### ISSUE THREAD

**Issue by volodya-lombrozo** [objectionary/hone-maven-plugin#65]
1. I pulled the project:

```
git clone git@github.com:objectionary/hone-maven-plugin.git
```

2. Then switched to the `58` branch: https://github.com/objectionary/hone-maven-plugin/tree/58
3. Then removed `@Disabled` annotation from the `OptimizeMojoTest#optimizesJnaClasses` 
4. Run this test from command line: `mvn test -Dtest=org.eolang.hone.OptimizeMojoTest#optimizesJnaClasses`
5. I got the following logs:

```
[INFO] --- surefire:3.3.1:test (default-test) @ hone-maven-plugin ---
[INFO] Tests will run in random order. To reproduce ordering use flag -Dsurefire.runOrder.random.seed=780997909236184
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.eolang.hone.OptimizeMojoTest
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [org.slf4j.reload4j.Reload4jServiceProvider@6a6cb05c]
SLF4J(W): Found provider [MavenSlf4j()]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [org.slf4j.reload4j.Reload4jServiceProvider@6a6cb05c]

```

Then the process stays in idle state. I don't know that is going on and how long I should wait. So, I waited ~ `5 min` and stopped the test (`Ctrl-C`).  Could you please add logs or to remove this tests. I believe unit test should take maximum `1 sec`, not  minutes. Moreover this test is silent, that confuses a lot.

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you help here, please?

**Comment by yegor256** [comment]
@volodya-lombrozo try again, please, the same branch (I merged master into it)

**Comment by volodya-lombrozo** [comment]
@yegor256 Now I'm getting this (relatively fast):
```java
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.eolang.hone.OptimizeMojoTest
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [org.slf4j.reload4j.Reload4jServiceProvider@6a6cb05c]
SLF4J(W): Found provider [MavenSlf4j()]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [org.slf4j.reload4j.Reload4jServiceProvider@6a6cb05c]
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 9.912 s <<< FAILURE! -- in org.eolang.hone.OptimizeMojoTest
[ERROR] org.eolang.hone.OptimizeMojoTest.optimizesJnaClasses(Path, String) -- Time elapsed: 9.877 s <<< FAILURE!
java.lang.AssertionError: 
the build must be successful
Expected: (a string containing "BUILD SUCCESS" and not a string containing "BUILD FAILURE")
     but: a string containing "BUILD SUCCESS" was "[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test 0.0.0
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ test ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /private/var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/junit9978005612697966655/src/main/resources
[INFO] 
[INFO] --- compiler:3.13.0:compile (default-compile) @ test ---
[INFO] No sources to compile
[INFO] 
[INFO] --- hone:1.0-SNAPSHOT:build (default) @ test ---
[INFO] >> #1 [internal] load build definition from Dockerfile
[INFO] >> #1 sha256:2ffa5cf4a1ab52eb469cd0becbd1b5a1adf7e4c150a8970427bbcf0b5e69428b
[INFO] >> #1 transferring dockerfile: 2.54kB 0.0s done
[INFO] >> #1 DONE 0.1s
[INFO] >> 
[INFO] >> #2 [internal] load .dockerignore
[INFO] >> #2 sha256:4748995d1e76a994ad08982161b088c59f58c3577bb0a12eb2b4cde5b781ccad
[INFO] >> #2 transferring context: 2B done
[INFO] >> #2 DONE 0.1s
[INFO] >> 
[INFO] >> #3 [internal] load metadata for docker.io/library/ubuntu:24.04
[INFO] >> #3 sha256:a47574ef4681921977ea75d38c5c3c7149a089de77020dc570bd4ffa2650961c
[INFO] >> #3 ...
[INFO] >> 
[INFO] >> #4 [auth] library/ubuntu:pull token for registry-1.docker.io
[INFO] >> #4 sha256:8bd30106b81c85723f186942446fee89122266278440f6edce91774037c97d17
[INFO] >> #4 DONE 0.0s
[INFO] >> 
[INFO] >> #3 [internal] load metadata for docker.io/library/ubuntu:24.04
[INFO] >> #3 sha256:a47574ef4681921977ea75d38c5c3c7149a089de77020dc570bd4ffa2650961c
[INFO] >> #3 DONE 2.2s
[INFO] >> 
[INFO] >> #5 [ 1/15] FROM docker.io/library/ubuntu:24.04@sha256:99c35190e22d294cdace2783ac55effc69d32896daaa265f0bbedbcde4fbe3e5
[INFO] >> #5 sha256:ab40d61598fde1a8a8471b4c483776064ed147aea6b716b38371a2658cc4bf08
[INFO] >> #5 resolve docker.io/library/ubuntu:24.04@sha256:99c35190e22d294cdace2783ac55effc69d32896daaa265f0bbedbcde4fbe3e5 0.0s done
[INFO] >> #5 DONE 0.0s
[INFO] >> 
[INFO] >> #20 [internal] load build context
[INFO] >> #20 sha256:e876ae62c7124310b210aa3b6e62dab801b3c8d352a3cb4cb9562616409c26d5
[INFO] >> #20 transferring context: 16.86kB 0.0s done
[INFO] >> #20 DONE 0.0s
[INFO] >> 
[INFO] >> #18 [14/15] COPY entry.sh ./
[INFO] >> #18 sha256:e325ab1d992f2fabc187a15cde54a64bfc37024064740e1603fbec9702feb1a8
[INFO] >> #18 CACHED
[INFO] >> 
[INFO] >> #10 [ 6/15] COPY install-ghc.sh ./
[INFO] >> #10 sha256:61e891bbe9c0248aafa9c30f86f7f9f37c8ce617521177b78ba83ebe0b0eabbd
[INFO] >> #10 CACHED
[INFO] >> 
[INFO] >> #7 [ 3/15] WORKDIR /hone
[INFO] >> #7 sha256:b150d34b6cd2f4b6dcf2d58edef31d5e2900af12e1f6dd59ab4b1fd4c9ef3194
[INFO] >> #7 CACHED
[INFO] >> 
[INFO] >> #17 [13/15] RUN mvn --update-snapshots --batch-mode --strict-checksums --errors eo:help jeo:help
[INFO] >> #17 sha256:596738b23a6bd1f0ed9d94cb096b7d0816724158db917ac5961dcf09510e9105
[INFO] >> #17 CACHED
[INFO] >> 
[INFO] >> #6 [ 2/15] RUN apt-get update   && apt-get install -y --no-install-recommends     ca-certificates=*     curl=*     dpkg-dev=*     g++=*     gcc=*     git=*     gnupg=*     libc6-dev=*     libffi-dev=*     libgmp-dev=*     libnuma-dev=*     make=*     netbase=*     openjdk-21-jdk-headless=*     tree=*     wget=*     xz-utils=*     zlib1g-dev=*   && rm -rf /var/lib/apt/lists/*
[INFO] >> #6 sha256:578238e3baffa8195a3e0cc6370251683bebc8cc54e2307d3538a2231f1dcf08
[INFO] >> #6 CACHED
[INFO] >> 
[INFO] >> #8 [ 4/15] COPY install-maven.sh ./
[INFO] >> #8 sha256:526cf02d397c1e832940bf9c33208b4f4dafc28d9541c7a2f0d07e983949810e
[INFO] >> #8 CACHED
[INFO] >> 
[INFO] >> #12 [ 8/15] COPY install-stack.sh ./
[INFO] >> #12 sha256:1c5f4fcf945ea4721b712a6fca86f2c2abe8f063318af32786bdf129ebbd1300
[INFO] >> #12 CACHED
[INFO] >> 
[INFO] >> #14 [10/15] COPY install-normalizer.sh ./
[INFO] >> #14 sha256:e41eb695cf55ef46802013f8b3406d7c750cdd62fc2d00c13fa464af114c672a
[INFO] >> #14 CACHED
[INFO] >> 
[INFO] >> #11 [ 7/15] RUN chmod a+x install-ghc.sh   && ./install-ghc.sh
[INFO] >> #11 sha256:9ded1881e299e6a54367bf5f6e23b8885cd235dfa6d2e958cee5bda3b582c358
[INFO] >> #11 CACHED
[INFO] >> 
[INFO] >> #13 [ 9/15] RUN chmod a+x install-stack.sh   && ./install-stack.sh   && stack config set system-ghc --global true
[INFO] >> #13 sha256:30e90ef2d557c78334141b6bd548f357bab80b2a3122cd917961e19c8d7135f7
[INFO] >> #13 CACHED
[INFO] >> 
[INFO] >> #9 [ 5/15] RUN chmod a+x install-maven.sh   && ./install-maven.sh
[INFO] >> #9 sha256:ae8d0a0a3d969e492f0b9294e7a4f21652590b933623b7407e9abd5e51938122
[INFO] >> #9 CACHED
[INFO] >> 
[INFO] >> #15 [11/15] RUN chmod a+x install-normalizer.sh   && ./install-normalizer.sh
[INFO] >> #15 sha256:ea6aba85ccba822e23a0cb6e02731ad3bbe512ef6deddbafe78de0255216eb38
[INFO] >> #15 CACHED
[INFO] >> 
[INFO] >> #16 [12/15] COPY in-docker-pom.xml pom.xml
[INFO] >> #16 sha256:bb8fd45c9579f94d6d19e26cd2fda7b957d3e35e8f49d1500ed944b4826a5978
[INFO] >> #16 CACHED
[INFO] >> 
[INFO] >> #19 [15/15] COPY rules/ ./rules/
[INFO] >> #19 sha256:10d020cfba099d1872788db508f616c5f4ae5f7954cbaa542169291e50d02d77
[INFO] >> #19 CACHED
[INFO] >> 
[INFO] >> #21 exporting to image
[INFO] >> #21 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
[INFO] >> #21 exporting layers done
[INFO] >> #21 writing image sha256:4b064d024f5c30834cec76f6e3f97651f06f70f733cfdc61e5f0610b3059e9a0 0.0s done
[INFO] >> #21 naming to docker.io/library/x3fdcf224d5c61df4 done
[INFO] >> #21 DONE 0.0s
[INFO] >> 
[INFO] >> Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
[INFO] + docker build --pull --tag x3fdcf224d5c61df4 /var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/tmp669852296467564223 -> 0x0000 in 4s
[INFO] 
[INFO] --- hone:1.0-SNAPSHOT:optimize (default) @ test ---
[INFO] >> standard_init_linux.go:228: exec user process caused: permission denied
[INFO] + docker run --rm --volume /private/var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/junit9978005612697966655/target:/target --env TARGET=/target --user 501:20 --env RULES=none.yml thirty-three.yml x3fdcf224d5c61df4 -> 0x0001 in 2s
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.100 s
[INFO] Finished at: 2024-10-30T12:24:35+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eolang:hone-maven-plugin:1.0-SNAPSHOT:optimize (default) on project test: java.io.IOException: Failed to execute docker, code=1 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eolang:hone-maven-plugin:1.0-SNAPSHOT:optimize (default) on project test: java.io.IOException: Failed to execute docker, code=1
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: java.io.IOException: Failed to execute docker, code=1
    at org.eolang.hone.AbstractMojo.execute (AbstractMojo.java:80)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.io.IOException: Failed to execute docker, code=1
    at org.eolang.hone.Docker.fire (Docker.java:106)
    at org.eolang.hone.Docker.exec (Docker.java:85)
    at org.eolang.hone.OptimizeMojo.exec (OptimizeMojo.java:151)
    at org.eolang.hone.AbstractMojo.execute (AbstractMojo.java:78)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.eolang.hone.OptimizeMojoTest.optimizeIt(OptimizeMojoTest.java:269)
	at org.eolang.hone.OptimizeMojoTest.lambda$optimizesJnaClasses$4(OptimizeMojoTest.java:240)
	at com.yegor256.farea.Farea.together(Farea.java:180)
	at org.eolang.hone.OptimizeMojoTest.optimizesJnaClasses(OptimizeMojoTest.java:189)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   OptimizeMojoTest.optimizesJnaClasses:189->lambda$optimizesJnaClasses$4:240->optimizeIt:269 the build must be successful
Expected: (a string containing "BUILD SUCCESS" and not a string containing "BUILD FAILURE")
     but: a string containing "BUILD SUCCESS" was "[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test 0.0.0
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- resources:3.3.1:resources (default-resources) @ test ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /private/var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/junit9978005612697966655/src/main/resources
[INFO] 
[INFO] --- compiler:3.13.0:compile (default-compile) @ test ---
[INFO] No sources to compile
[INFO] 
[INFO] --- hone:1.0-SNAPSHOT:build (default) @ test ---
[INFO] >> #1 [internal] load build definition from Dockerfile
[INFO] >> #1 sha256:2ffa5cf4a1ab52eb469cd0becbd1b5a1adf7e4c150a8970427bbcf0b5e69428b
[INFO] >> #1 transferring dockerfile: 2.54kB 0.0s done
[INFO] >> #1 DONE 0.1s
[INFO] >> 
[INFO] >> #2 [internal] load .dockerignore
[INFO] >> #2 sha256:4748995d1e76a994ad08982161b088c59f58c3577bb0a12eb2b4cde5b781ccad
[INFO] >> #2 transferring context: 2B done
[INFO] >> #2 DONE 0.1s
[INFO] >> 
[INFO] >> #3 [internal] load metadata for docker.io/library/ubuntu:24.04
[INFO] >> #3 sha256:a47574ef4681921977ea75d38c5c3c7149a089de77020dc570bd4ffa2650961c
[INFO] >> #3 ...
[INFO] >> 
[INFO] >> #4 [auth] library/ubuntu:pull token for registry-1.docker.io
[INFO] >> #4 sha256:8bd30106b81c85723f186942446fee89122266278440f6edce91774037c97d17
[INFO] >> #4 DONE 0.0s
[INFO] >> 
[INFO] >> #3 [internal] load metadata for docker.io/library/ubuntu:24.04
[INFO] >> #3 sha256:a47574ef4681921977ea75d38c5c3c7149a089de77020dc570bd4ffa2650961c
[INFO] >> #3 DONE 2.2s
[INFO] >> 
[INFO] >> #5 [ 1/15] FROM docker.io/library/ubuntu:24.04@sha256:99c35190e22d294cdace2783ac55effc69d32896daaa265f0bbedbcde4fbe3e5
[INFO] >> #5 sha256:ab40d61598fde1a8a8471b4c483776064ed147aea6b716b38371a2658cc4bf08
[INFO] >> #5 resolve docker.io/library/ubuntu:24.04@sha256:99c35190e22d294cdace2783ac55effc69d32896daaa265f0bbedbcde4fbe3e5 0.0s done
[INFO] >> #5 DONE 0.0s
[INFO] >> 
[INFO] >> #20 [internal] load build context
[INFO] >> #20 sha256:e876ae62c7124310b210aa3b6e62dab801b3c8d352a3cb4cb9562616409c26d5
[INFO] >> #20 transferring context: 16.86kB 0.0s done
[INFO] >> #20 DONE 0.0s
[INFO] >> 
[INFO] >> #18 [14/15] COPY entry.sh ./
[INFO] >> #18 sha256:e325ab1d992f2fabc187a15cde54a64bfc37024064740e1603fbec9702feb1a8
[INFO] >> #18 CACHED
[INFO] >> 
[INFO] >> #10 [ 6/15] COPY install-ghc.sh ./
[INFO] >> #10 sha256:61e891bbe9c0248aafa9c30f86f7f9f37c8ce617521177b78ba83ebe0b0eabbd
[INFO] >> #10 CACHED
[INFO] >> 
[INFO] >> #7 [ 3/15] WORKDIR /hone
[INFO] >> #7 sha256:b150d34b6cd2f4b6dcf2d58edef31d5e2900af12e1f6dd59ab4b1fd4c9ef3194
[INFO] >> #7 CACHED
[INFO] >> 
[INFO] >> #17 [13/15] RUN mvn --update-snapshots --batch-mode --strict-checksums --errors eo:help jeo:help
[INFO] >> #17 sha256:596738b23a6bd1f0ed9d94cb096b7d0816724158db917ac5961dcf09510e9105
[INFO] >> #17 CACHED
[INFO] >> 
[INFO] >> #6 [ 2/15] RUN apt-get update   && apt-get install -y --no-install-recommends     ca-certificates=*     curl=*     dpkg-dev=*     g++=*     gcc=*     git=*     gnupg=*     libc6-dev=*     libffi-dev=*     libgmp-dev=*     libnuma-dev=*     make=*     netbase=*     openjdk-21-jdk-headless=*     tree=*     wget=*     xz-utils=*     zlib1g-dev=*   && rm -rf /var/lib/apt/lists/*
[INFO] >> #6 sha256:578238e3baffa8195a3e0cc6370251683bebc8cc54e2307d3538a2231f1dcf08
[INFO] >> #6 CACHED
[INFO] >> 
[INFO] >> #8 [ 4/15] COPY install-maven.sh ./
[INFO] >> #8 sha256:526cf02d397c1e832940bf9c33208b4f4dafc28d9541c7a2f0d07e983949810e
[INFO] >> #8 CACHED
[INFO] >> 
[INFO] >> #12 [ 8/15] COPY install-stack.sh ./
[INFO] >> #12 sha256:1c5f4fcf945ea4721b712a6fca86f2c2abe8f063318af32786bdf129ebbd1300
[INFO] >> #12 CACHED
[INFO] >> 
[INFO] >> #14 [10/15] COPY install-normalizer.sh ./
[INFO] >> #14 sha256:e41eb695cf55ef46802013f8b3406d7c750cdd62fc2d00c13fa464af114c672a
[INFO] >> #14 CACHED
[INFO] >> 
[INFO] >> #11 [ 7/15] RUN chmod a+x install-ghc.sh   && ./install-ghc.sh
[INFO] >> #11 sha256:9ded1881e299e6a54367bf5f6e23b8885cd235dfa6d2e958cee5bda3b582c358
[INFO] >> #11 CACHED
[INFO] >> 
[INFO] >> #13 [ 9/15] RUN chmod a+x install-stack.sh   && ./install-stack.sh   && stack config set system-ghc --global true
[INFO] >> #13 sha256:30e90ef2d557c78334141b6bd548f357bab80b2a3122cd917961e19c8d7135f7
[INFO] >> #13 CACHED
[INFO] >> 
[INFO] >> #9 [ 5/15] RUN chmod a+x install-maven.sh   && ./install-maven.sh
[INFO] >> #9 sha256:ae8d0a0a3d969e492f0b9294e7a4f21652590b933623b7407e9abd5e51938122
[INFO] >> #9 CACHED
[INFO] >> 
[INFO] >> #15 [11/15] RUN chmod a+x install-normalizer.sh   && ./install-normalizer.sh
[INFO] >> #15 sha256:ea6aba85ccba822e23a0cb6e02731ad3bbe512ef6deddbafe78de0255216eb38
[INFO] >> #15 CACHED
[INFO] >> 
[INFO] >> #16 [12/15] COPY in-docker-pom.xml pom.xml
[INFO] >> #16 sha256:bb8fd45c9579f94d6d19e26cd2fda7b957d3e35e8f49d1500ed944b4826a5978
[INFO] >> #16 CACHED
[INFO] >> 
[INFO] >> #19 [15/15] COPY rules/ ./rules/
[INFO] >> #19 sha256:10d020cfba099d1872788db508f616c5f4ae5f7954cbaa542169291e50d02d77
[INFO] >> #19 CACHED
[INFO] >> 
[INFO] >> #21 exporting to image
[INFO] >> #21 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
[INFO] >> #21 exporting layers done
[INFO] >> #21 writing image sha256:4b064d024f5c30834cec76f6e3f97651f06f70f733cfdc61e5f0610b3059e9a0 0.0s done
[INFO] >> #21 naming to docker.io/library/x3fdcf224d5c61df4 done
[INFO] >> #21 DONE 0.0s
[INFO] >> 
[INFO] >> Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
[INFO] + docker build --pull --tag x3fdcf224d5c61df4 /var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/tmp669852296467564223 -> 0x0000 in 4s
[INFO] 
[INFO] --- hone:1.0-SNAPSHOT:optimize (default) @ test ---
[INFO] >> standard_init_linux.go:228: exec user process caused: permission denied
[INFO] + docker run --rm --volume /private/var/folders/w0/69c9m3f92t9bbcdw8ptgfjyw0000gn/T/junit9978005612697966655/target:/target --env TARGET=/target --user 501:20 --env RULES=none.yml thirty-three.yml x3fdcf224d5c61df4 -> 0x0001 in 2s
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.100 s
[INFO] Finished at: 2024-10-30T12:24:35+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eolang:hone-maven-plugin:1.0-SNAPSHOT:optimize (default) on project test: java.io.IOException: Failed to execute docker, code=1 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eolang:hone-maven-plugin:1.0-SNAPSHOT:optimize (default) on project test: java.io.IOException: Failed to execute docker, code=1
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: java.io.IOException: Failed to execute docker, code=1
    at org.eolang.hone.AbstractMojo.execute (AbstractMojo.java:80)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.io.IOException: Failed to execute docker, code=1
    at org.eolang.hone.Docker.fire (Docker.java:106)
    at org.eolang.hone.Docker.exec (Docker.java:85)
    at org.eolang.hone.OptimizeMojo.exec (OptimizeMojo.java:151)
    at org.eolang.hone.AbstractMojo.execute (AbstractMojo.java:78)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
"
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.931 s
[INFO] Finished at: 2024-10-30T12:24:36+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.3.1:test (default-test) on project hone-maven-plugin: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/lombrozo/Workspace/EOlang/hone-maven-plugin/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

```

**Comment by yegor256** [comment]
@volodya-lombrozo this is a separate issue, better report as new one

**Comment by volodya-lombrozo** [comment]
@yegor256 Ok. I removed the image related to `hone-maven-plugin`, restarted test again and got this stacktrace: https://github.com/objectionary/hone-maven-plugin/issues/65#issuecomment-2446309475

As for the current issue - I can't check it before [this](https://github.com/objectionary/hone-maven-plugin/issues/65#issuecomment-2446309475) will be solved

**Comment by volodya-lombrozo** [comment]
@yegor256 This issue blocked by https://github.com/objectionary/hone-maven-plugin/issues/67

**Comment by volodya-lombrozo** [comment]
@yegor256 I tried to run this test again by using the same command:

```
mvn test -Dtest=org.eolang.hone.OptimizeMojoTest#optimizesJnaClasses
```

and I faced with the same problem again. I saw this:

```bash

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.eolang.hone.OptimizeMojoTest
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [org.slf4j.reload4j.Reload4jServiceProvider@408d971b]
SLF4J(W): Found provider [MavenSlf4j()]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [org.slf4j.reload4j.Reload4jServiceProvider@408d971b]
```

for 42 minutes. Nothing was printed. I was just sitting and hoping that something is happening. It's frustrating. Only after 42 minutes I saw a huge log that explains the problem.
```
....
huge log here
....
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  42:52 min
[INFO] Finished at: 2024-10-30T23:09:28+03:00
[INFO] ------------------------------------------------------------------------
```

I understand what's going on under the hood, but still believe, that it's better to report at least some progress. Otherwise, I don't think anybody will wait for so long.

**Comment by yegor256** [comment]
@volodya-lombrozo try again, after c399361

**Comment by volodya-lombrozo** [comment]
@yegor256 It's a bit better, of course, but I would add some context to this messages, some sort of progress:
```
SLF4J(W): Class path contains multiple SLF4J providers.
SLF4J(W): Found provider [org.slf4j.reload4j.Reload4jServiceProvider@74e52303]
SLF4J(W): Found provider [MavenSlf4j()]
SLF4J(W): See https://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J(I): Actual provider is of type [org.slf4j.reload4j.Reload4jServiceProvider@74e52303]
09:55:35 [WARN] org.eolang.hone.MayBeSlow: The test optimizesJnaClasses(Path, String) may take longer than a minute; if you want to see the full output of it, set the logging level to "DEBUG" for the "com.yegor256.farea" logging facility, in the "src/test/resources/log4j.properties" file
09:55:40 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (5s), please wait...
09:55:45 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (10s), please wait...
09:55:50 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (15s), please wait...
09:55:55 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (20s), please wait...
09:56:00 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (25s), please wait...
09:56:05 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (30s), please wait...
09:56:10 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (35s), please wait...
09:56:15 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (40s), please wait...
09:56:20 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (45s), please wait...
09:56:25 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (50s), please wait...
09:56:30 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (55s), please wait...
09:56:35 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (1min), please wait...
09:56:40 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (1min), please wait...
09:56:45 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (1min), please wait...
09:56:50 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (1min), please wait...
09:56:55 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (1min), please wait...
09:57:00 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (1min), please wait...
09:57:05 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (2min), please wait...
09:57:10 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (2min), please wait...
09:57:15 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (2min), please wait...
09:57:20 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (2min), please wait...
09:57:25 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (2min), please wait...
09:57:30 [WARN] org.eolang.hone.MayBeSlow$1: We're still running the test (2min), please wait...
```

Ok, maybe it's  enough for now.
